dcos_test_utils/marathon.py

```diff
@@ -424,3 +396,25 @@ def deploy_and_cleanup(self, app_definition, timeout=120, check_health=True, ign
     def deploy_pod_and_cleanup(self, pod_definition, timeout=300):
         yield self.deploy_pod(pod_definition, timeout=timeout)
         self.destroy_pod(pod_definition['id'], timeout)
+
+    def purge(self):
+        """ Deletes all applications, all pods, and then waits
+        indefinitely for any deployments to finish
+        """
+        for app in self.get('v2/apps').json()['apps']:
+            log.info('Purging application: {}'.format(app['id']))
+            self.delete('v2/apps' + app['id'])
+        for pod in self.get('v2/pods').json():
+            log.info('Deleting pod: {}'.format(pod['id']))
+            self.delete('v2/pods' + pod['id'])
```

Comment:
@kensipe Wrong repo :P I think you are thinking of this code being: https://github.com/dcos/dcos-cli/blob/master/dcos/marathon.py

This self.delete is a much more minimal object that is making a requests.delete call with an auth header attached. That being said, I agree and I think we should duplicate the logic here (https://github.com/dcos/dcos-cli/blob/master/dcos/marathon.py#L786)